### PR TITLE
Moving has_photos scope to PhotoCapable

### LINKED ISSUE #1294: move has_photos scope to PhotoCapable

### PR DESCRIPTION
**Pull request by Br3nda**
This moves some repeating code to photocapable.

fixes #1294

### CODE REVIEW

**Comment by Br3nda** [comment]
I've removed the default scope from Harvest. It causes an ORDER clause to be added in queries even when not selecting from that table. E.g. if we 're looking for PlantParts that have a Harvest

File: app/models/harvest.rb [hunk 1]
```diff
@@ -9,7 +9,6 @@ class Harvest < ActiveRecord::Base
   belongs_to :plant_part
   belongs_to :planting
 
-  default_scope { order('created_at DESC') }
```
Comment (CloCkWeRX) [inline app/models/harvest.rb]: "Harvest ordering lists most recent harvests first" seems upset by this; may need corresponding changes to various controllers, specs, views

**Comment by Br3nda** [comment]
The order ends up in crop's `has_many :plant_parts, through: :harvests`
Yeah, this needs more time than i could give in my lunch break.

File: app/models/harvest.rb (hunk 1 above)
Comment (Br3nda) [inline app/models/harvest.rb]: The default scopes i've seen in other projects are e.g. `where(deleted: false)` etc. 

Growstuff has a log of query calls in the views too - where I've usually seen them in the controller with order and pagination controlled there.

Comment (Br3nda) [inline app/models/harvest.rb]: I've added this default scope back in, and added a `reorder()` to plant_parts.harvests. this does what you'd expect unscoped to do.

Comment (Br3nda) [inline app/models/harvest.rb]: Related: https://github.com/rails/rails/issues/5701
Unscoped doesn't take away the default scope's order by, but reorder did the trick.

**Pull request merged**